omod/src/main/java/org/openmrs/web/dwr/DWRPatientService.java

```diff
@@ -335,7 +335,29 @@ public PatientListItem getPatient(Integer patientId) {
 		}
 		return pli;
 	}
-	
+
+	/**
+	 * find all the patients with the given patient identifiers (identifiers)
+	 *
+	 * @param identifiers
+	 * @return list of patientListItems
+	 */
+	public Vector<Object> findPatientsByIdentifier(String[] identifiers) {
+		Vector<Object> patientList = new Vector<>();
+		for (String identifier : identifiers) {
+			try {
+				List<Patient> patients = Context.getPatientService().getPatients(identifier, false, 0, 1);
+				if(patients.size() > 0){
+					patientList.add(new PatientListItem(patients.get(0)));
+				}
+			} catch (Exception e) {
+				log.error("Error while searching for patients", e);
```

Comment:
We have implemented the given feedback.
To reduce search time, we have used `getPatientIdentifiers()` instead of `getPatients` to find a patient.